SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp

```diff
@@ -1,7 +1,382 @@
+	AddComponentToView(ExpectedView, TestEntityId, TestComponentId, OtherTestComponentValue);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::UPDATE)
+		.AddComponentRefreshed(TestEntityId, ComponentUpdate(TestComponentId),
+							   CreateTestComponentData(TestComponentId, OtherTestComponentValue));
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_view_WHEN_entity_remove_and_add_THEN_no_entity_flag)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+	AddEntityToView(InputView, TestEntityId);
+
+	EntityComponentOpListBuilder OpListBuilder;
+	OpListBuilder.RemoveEntity(TestEntityId).AddEntity(TestEntityId);
+	SetFromOpList(InputDelta, InputView, OpListBuilder);
+
+	EntityView ExpectedView;
+	AddEntityToView(ExpectedView, TestEntityId);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::UPDATE);
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_empty_view_WHEN_add_remove_add_THEN_entity_in_view_and_delta)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+
+	EntityComponentOpListBuilder OpListBuilder;
+	OpListBuilder.AddEntity(TestEntityId).RemoveEntity(TestEntityId).AddEntity(TestEntityId);
+	SetFromOpList(InputDelta, InputView, OpListBuilder);
+
+	EntityView ExpectedView;
+	AddEntityToView(ExpectedView, TestEntityId);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::EntityChangeType::ADD);
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_empty_view_WHEN_add_entity_add_component_THEN_entity_and_component_in_view_and_delta)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+
+	EntityComponentOpListBuilder OpListBuilder;
+	OpListBuilder.AddEntity(TestEntityId).AddComponent(TestEntityId, CreateTestComponentData(TestComponentId, TestComponentValue));
+	SetFromOpList(InputDelta, InputView, OpListBuilder);
+
+	EntityView ExpectedView;
+	AddEntityToView(ExpectedView, TestEntityId);
+	AddComponentToView(ExpectedView, TestEntityId, TestComponentId, TestComponentValue);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::EntityChangeType::ADD)
+		.AddComponentAdded(TestEntityId, CreateTestComponentData(TestComponentId, TestComponentValue));
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", AreEquivalent(InputView, ExpectedView));
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_and_component_in_view_WHEN_remove_entity_THEN_empty_view_remove_ops_in_delta)
```

Comment:
This is an odd one. It will work but partially by implementation rather than by contract. The assumption is that you never get a remove entity without a remove component first.

Reply:
I think just remove the component first and it's good.